mobs/nakki_boss/FightDirector.cs

```diff
@@ -33,10 +37,7 @@ public Node2D StartPosition {
 	}
 	private Node2D? _startPosition;
 
-
-	private readonly List<LilypadAttackStats> _attacksWaitingForNakkiSignal = [];
-	private readonly List<Timer> _delayedAttacks = [];
-
+	private CancellationTokenSource _lilypadAttackCancelSource = new();
```

Comment:
This is where the honeymoon ended. Cancelling tasks is a huge PITA, but I assume these still result in cleaner code than with timers-based state machines.